src/test/java/com/microsoft/sqlserver/jdbc/connection/PoolingTest.java

```diff
@@ -139,4 +145,52 @@ public void testConnectionPoolClientConnectionId() throws SQLException {
 
         assertEquals(Id1, Id2, "ClientConnection Ids from pool are not the same.");
     }
+    
+    @Test
+    public void testHikariCP() throws SQLException {
+        HikariConfig config = new HikariConfig();
+        config.setJdbcUrl(connectionString);
+        HikariDataSource ds = new HikariDataSource(config);
+
+        connect(ds);
```

Comment:
instead of doing `if (ttEndCount - ttStartCount > 2) { throw exception}`, I guess only verify ttEndCount is at least 1 is sufficient. In the for loop, if we increase the termination (i.g. to 100000), it may create more than 2 timeout threads.